neo4j/internal/packstream/packer.go

```diff
@@ -262,3 +262,105 @@ func (p *Packer) checkOverflowInt(i uint64) {
 		p.err = &OverflowError{msg: "Trying to pack uint64 that doesn't fit into int64"}
 	}
 }
+
+func (p *Packer) VectorFloat64(vec []float64) {
+	p.StructHeader('V', 2)
+	p.Bytes([]byte{0xc1}) // FLOAT_64 marker
+
+	if len(vec) == 0 {
+		p.Bytes([]byte{}) // empty values
+		return
+	}
+
+	totalSize := len(vec) * 8
+	values := make([]byte, totalSize)
+	for i, v := range vec {
+		binary.BigEndian.PutUint64(values[i*8:], math.Float64bits(v))
+	}
```

Comment:
As suggested in the hydrator: no need to zero the memory first.

This saves around 10% of time on my machine.

```suggestion
	values := make([]byte, 0, totalSize)
	for _, v := range vec {
		values = binary.BigEndian.AppendUint64(values, math.Float64bits(v))
	}
```

Reply:
Here's another sketched suggestion that's even another 10-ish% faster on machine. Making this pretty is left as an exercise to the reader :stuck_out_tongue_winking_eye: 

```go
func (p *Packer) VectorFloat64(vec []float64) {
	p.StructHeader('V', 2)
	p.Bytes([]byte{0xc1}) // FLOAT_64 marker

	if len(vec) == 0 {
		p.Bytes([]byte{}) // empty values
		return
	}

	totalSize := len(vec) * 8
	err := p.bytesHeader(int64(totalSize))
	if err != nil {
		p.err = err
		return
	}

	neededBuf := totalSize
	if cap(p.buf)-len(p.buf) < neededBuf {
		newBuf := make([]byte, len(p.buf), max(cap(p.buf)*2, len(p.buf)+neededBuf))
		copy(newBuf, p.buf)
		p.buf = newBuf
	}

	for _, v := range vec {
		p.buf = binary.BigEndian.AppendUint64(p.buf, math.Float64bits(v))
	}
}

func (p *Packer) Bytes(b []byte) {
	err := p.bytesHeader(int64(len(b)))
	if err != nil {
		p.err = err
		return
	}
	p.buf = append(p.buf, b...)
}

func (p *Packer) bytesHeader(l int64) error {
	hdr := make([]byte, 0, 1+4)
	switch {
	case l < 0x100:
		hdr = append(hdr, 0xcc, byte(l))
	case l < 0x10000:
		hdr = hdr[:1+2]
		hdr[0] = 0xcd
		binary.BigEndian.PutUint16(hdr[1:], uint16(l))
	case l < 0x100000000:
		hdr = hdr[:1+4]
		hdr[0] = 0xce
		binary.BigEndian.PutUint32(hdr[1:], uint32(l))
	default:
		return &OverflowError{msg: fmt.Sprintf("Trying to pack too large byte array of size %d", l)}
	}
	p.buf = append(p.buf, hdr...)
	return nil
}
```

Reply:
Brilliant, I refactored the Vector* functions to fit this pattern 👍 confirmed the performance increase too.